app/models/batch_context.rb

```diff
@@ -63,6 +63,10 @@ def progress_log_file
     @progress_log_file ||= File.join(output_dir, "#{project_name}_progress.yml")
   end
 
+  def progress_log_file_exists?
+    progress_log_file && File.exist?(progress_log_file)
+  end
+
```

Comment:
just moves this method from the `JobRun` class to the `BatchContext` class and then delegated here (since we now want to interrogate it at the level of a BatchContext